core/domain/question_domain_test.py

```diff
@@ -13,13 +13,105 @@
 # limitations under the License.
 
 """Tests for question domain objects."""
+import datetime
 
 from core.domain import question_domain
+from core.domain import state_domain
 from core.tests import test_utils
 import feconf
 import utils
 
 
+class QuestionChangeTest(test_utils.GenericTestBase):
+    """Test for Question Change object."""
+
+    def test_to_dict(self):
+        """Test to verify to_dict method of the Question Change object."""
+        expected_object_dict = {
+            'cmd': 'update_question_property',
+            'property_name': 'question_state_data',
+            'new_value': 'new_value',
+            'old_value': 'old_value',
+        }
+
+        change_dict = {
+            'cmd': 'update_question_property',
+            'property_name': 'question_state_data',
+            'new_value': 'new_value',
+            'old_value': 'old_value',
+        }
+        observed_object = question_domain.QuestionChange(
+            change_dict=change_dict,
+        )
+
+        self.assertEqual(expected_object_dict, observed_object.to_dict())
+
+    def test_change_dict_without_cmd(self):
+        """Test to verify __init__ method of the Question Change object
+        when change_dict is without cmd key.
+        """
+        self.assertRaises(
+            Exception,
+            callableObj=question_domain.QuestionChange,
+            change_dict={}
+        )
+
+    def test_change_dict_with_wrong_cmd(self):
+        """Test to verify __init__ method of the Question Change object
+        when change_dict is with wrong cmd value.
+        """
+        self.assertRaises(
+            Exception,
+            callableObj=question_domain.QuestionChange,
+            change_dict={'cmd': 'wrong', }
+        )
+
+    def test_update_question_property_with_wrong_property_name(self):
+        """Test to verify __init__ method of the Question Change object
+        when cmd is update_question_property and wrong property_name is given.
+        """
+        self.assertRaises(
+            Exception,
+            callableObj=question_domain.QuestionChange,
+            change_dict={
+                'cmd': 'update_question_property',
+                'property_name': 'wrong',
+            }
+        )
+
+    def test_create_new_fully_specified_question(self):
+        """Test to verify __init__ method of the Question Change object
+        when cmd is create_new_fully_specified_question.
+        """
+        change_dict = {
+            'cmd': 'create_new_fully_specified_question',
+            'question_dict': {},
+            'skill_id': 10,
```

Comment:
`skill_id` is a string, so can you give a dummy string here?

Reply:
Ok.